src/coreclr/src/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -218,6 +219,7 @@ protected virtual MethodInfo GetMethodImpl()
         public object? Target => GetTarget();
 
         // V1 API.
+        [RequiresUnreferencedCode("The target method might be removed")]
```

Comment:
Nit: is this message going to make sense with the context consumers of it have?  "removed" by what and when?  Also, punctuation?

Reply:
@vitek-karas @eerhardt What do you think about this?

We could also go around the "removed" reference and say "the target method might not be available". We should probably make sure similar language is used everywhere (might be good to capture whatever we decide in a guideline). I'll leave the decision to native speakers. Some of English the language nuances still escape me.

The dot at the end gets added when linker generates the full message from this. The full message looks like _"Calling 'X' which has `RequiresUnreferencedCodeAttribute` can break functionality when trimming application code. Y."_ Y is the message from here.

Reply:
I think it sounds better without the "removed", but I'm in the same boat as you - not a native speaker. I think we need a review of these messages from a "PM-like" person (and native speaker). I filed https://github.com/mono/linker/issues/1275 (if we ever get to it).

Reply:
Okay, I'll keep the wording for now since we already added similar wording with "removed" in the past pull request. We can review as part of the issue Vitek filed.

Reply:
> The dot at the end gets added when linker generates the full message from this.

That seems a surprising behavior. If someone new adds one of these attributes, I don't think they would know to leave off the period from the message.